crates/rpc/README.md

```diff
@@ -1,129 +1,112 @@
+- [GetBlockByNumber](#getblockbynumber)
+- [GetBlockHeaderByNumber](#getblockheaderbynumber)
+- [GetNotesById](#getnotesbyid)
+- [SubmitProvenTransaction](#submitproventransaction)
+- [SyncNotes](#syncnotes)
+- [SyncState](#syncstate)
+<!--toc:end-->
 
-## API
-
-The **RPC** serves connections using the [gRPC protocol](https://grpc.io) on a port, set in the previously mentioned configuration file.
-Here is a brief description of supported methods.
+---
 
 ### CheckNullifiers
 
 Gets a list of proofs for given nullifier hashes, each proof as a sparse Merkle Tree.
 
-**Parameters:**
+---
 
-- `nullifiers`: `[Digest]` – array of nullifier hashes.
+### CheckNullifiersByPrefix
 
-**Returns:**
+Returns a list of nullifiers that match the specified prefixes and are recorded in the node.
 
-- `proofs`: `[NullifierProof]` – array of nullifier proofs, positions correspond to the ones in request.
+---
 
-### GetBlockHeaderByNumber
+### GetAccountDetails
+
+Returns the latest state of an account with the specified ID.
+
+---
+
+### GetAccountProofs
 
-Retrieves block header by given block number, optionally alongside a Merkle path and the current chain length to validate its inclusion.
+Returns the latest state proofs of the specified accounts.
 
-**Parameters**
+---
 
-- `block_num`: `uint32` _(optional)_ – the block number of the target block. If not provided, the latest known block will be returned.
+### GetAccountStateDelta
 
-**Returns:**
+Returns delta of the account states in the range from `from_block_num` (exclusive) to `to_block_num` (inclusive).
 
-- `block_header`: `BlockHeader` – block header.
+---
 
 ### GetBlockByNumber
 
 Retrieves block data by given block number.
 
-**Parameters**
+---
 
-- `block_num`: `uint32` – the block number of the target block.
+### GetBlockHeaderByNumber
 
-**Returns:**
+Retrieves block header by given block number. Optionally, it also returns the MMR path and current chain length to
+authenticate the block's inclusion.
 
-- `block`: `Block` – block data encoded using [winter_utils::Serializable](https://github.com/facebook/winterfell/blob/main/utils/core/src/serde/mod.rs#L26) implementation for [miden_objects::block::Block](https://github.com/0xPolygonMiden/miden-base/blob/main/objects/src/block/mod.rs#L43).
+---
 
 ### GetNotesById
 
 Returns a list of notes matching the provided note IDs.
 
-**Parameters**
-
-- `note_ids`: `[NoteId]` - list of IDs of the notes we want to query.
+---
 
-**Returns**
+### SubmitProvenTransaction
 
-- `notes`: `[Note]` - List of notes matching the list of requested NoteIds.
+Submits proven transaction to the Miden network.
 
-### GetAccountDetails
+---
 
-Returns the latest state of an account with the specified ID.
+### SyncNotes
 
-**Parameters**
+Note synchronization request.
 
-- `account_id`: `AccountId` – account ID.
+Specifies note tags that client is interested in. The server will return the first block which contains a note matching
+`note_tags` or the chain tip.
```

Comment:
I think the second sentence should be more like: "The server will return the matching notes from the first block which contains a note matching `note_tags`."

I would also maybe make this description more detailed - similar to how we have it for `SyncState` endpoint (and probably propagate these details to the protobuf files as well).